v1/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerSchema.java

```diff
@@ -192,13 +202,30 @@ private static Type parseSpannerType(String spannerType, Dialect dialect) {
             return Type.json();
           }
           if (spannerType.startsWith("ARRAY")) {
-            // Substring "ARRAY<xxx>"
-            String spannerArrayType = spannerType.substring(6, spannerType.length() - 1);
-            Type itemType = parseSpannerType(spannerArrayType, dialect);
+            // Substring "ARRAY<xxx>" or substring "ARRAY<xxx>(vector_length=>yyy)"
+
+            String arrayElementType;
+            // Handle vector_length annotation
+            Matcher m = EMBEDDING_VECTOR_PATTERN.matcher(spannerType);
+            if (m.find()) {
+              arrayElementType = m.group(1);
+            }
+            else {
+              arrayElementType = spannerType.substring(6, spannerType.length() - 1);
```

Comment:
Nit: don't like substring here for the same reason as last PR, but I don't consider it to be a blocking issue.

Reply:
This is already existing code and beyond the scope of this change IMO.
The code that handles vector length uses regex patten matching.